wagtail/users/utils.py

```diff
@@ -25,26 +27,59 @@ def user_can_delete_user(current_user, user_to_delete):
     return True
 
 
-def get_gravatar_url(email, size=50):
-    default = "mp"
-    size = (
-        int(size) * 2
-    )  # requested at retina size by default and scaled down at point of use with css
+def get_gravatar_url(email, size=50, default_params={"d": "mp"}):
+    """
+    See https://gravatar.com/site/implement/images/ for Gravatar image options
+    Example usage:
+    ```python
+    # Basic usage
+    gravatar_url = get_gravatar_url('user@example.com')
+
+    # Customize size and default image
+    gravatar_url = get_gravatar_url(
+        'user@example.com',
+        size=100,
+        default_params={'d': 'robohash', 'f': 'y'}
+    )
+    Note: If any parameter in `default_params` also exists in the provider URL,
+    It will be overridden by the provider URL's query parameter.
+    ```
+    """
+
+    DEFAULT_GRAVATAR_PROVIDER_URL = "//www.gravatar.com/avatar"
+
+    params = {
+        **default_params,
+        # requested at retina size by default and scaled down at point of use with css
+        "s": int(size) * 2,
+    }
+
     gravatar_provider_url = getattr(
-        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", "//www.gravatar.com/avatar"
+        settings, "WAGTAIL_GRAVATAR_PROVIDER_URL", DEFAULT_GRAVATAR_PROVIDER_URL
     )
 
     if (not email) or (gravatar_provider_url is None):
         return None
 
-    email_bytes = email.lower().encode("utf-8")
-    hash = safe_md5(email_bytes, usedforsecurity=False).hexdigest()
-    gravatar_url = "{gravatar_provider_url}/{hash}?{params}".format(
-        gravatar_provider_url=gravatar_provider_url.rstrip("/"),
-        hash=hash,
-        params=urlencode({"s": size, "d": default}),
+    parsed_url = urlparse(gravatar_provider_url)
+
+    if parsed_url.query:
+        parsed_qs = parse_qs(parsed_url.query)
+        parsed_qs.pop("s", None)
+        parsed_qs.pop("size", None)
+        params.update(parsed_qs)
+
```

Comment:
Let's keep this simple, just prepare the params in one go, if there's no qs to parse, just parse an empty string.

No need to pop/remove `s` this way.

Do not worry about the size edge case, if the developer really wants to use `size`, they will probably know what they are doing and want to do it intentionally.

```suggestion
    params = {
        **default_params,
        **(parse_qs(parsed_url.query or '')),
        # requested at retina size by default and scaled down at point of use with css
        "s": int(size) * 2,
    }

```